pkg/cloudprovider/aws/instance.go

```diff
@@ -219,12 +219,16 @@ func (p *InstanceProvider) getInstances(ctx context.Context, ids []*string) ([]*
 		return nil, fmt.Errorf("expected %d instance(s), but got %d", len(ids), len(describedInstances))
 	}
 	instances := []*ec2.Instance{}
-	for _, instance := range describedInstances {
-		if len(aws.StringValue(instance.PrivateDnsName)) == 0 {
-			err = multierr.Append(err, fmt.Errorf("got instance %s but PrivateDnsName was not set", aws.StringValue(instance.InstanceId)))
-			continue
+	if injection.GetOptions(ctx).AWSNodeNameConvention == "resource-name" {
```

Comment:
Cleaner to shortcircuit instead of if/else.
```
if injection.GetOptions(ctx).AWSNodeNameConvention == "resource-name" {
  return instances, nil
}
```

Reply:
You wouldn't want the constants in the options package instead?

Reply:
You're all over it :). Wasn't looking at the code, but you're 100% correct.